packages/react-data-grid/src/common/editors/EditorContainer.tsx

```diff
@@ -1,274 +1,247 @@
 type SharedInteractionMasksState = Pick<InteractionMasksState, 'firstEditorKeyPress'>;
 
-type ValueType<R> = R[keyof R];
-
-export interface Props<R, K extends keyof R> extends SharedInteractionMasksProps<R, K>, SharedInteractionMasksState, Omit<Dimension, 'zIndex'> {
+export interface EditorContainerProps<R, RK extends keyof R, CK extends keyof R> extends SharedInteractionMasksProps<R, RK>, SharedInteractionMasksState, Omit<Dimension, 'zIndex'> {
   rowIdx: number;
   rowData: R;
-  value: ValueType<R>;
-  column: CalculatedColumn<R>;
-  onGridKeyDown?(e: KeyboardEvent): void;
-  onCommit(e: CommitEvent<R>): void;
+  value: R[CK];
+  column: CalculatedColumn<R, unknown, CK>;
+  onCommit(e: CommitEvent<R, { [k: string]: R[CK] }>): void;
   onCommitCancel(): void;
 }
 
-interface State {
-  isInvalid: boolean;
-}
-
-export default class EditorContainer<R, K extends keyof R> extends React.Component<Props<R, K>, State> {
-  static displayName = 'EditorContainer';
+function getInitialValue<R, CK extends keyof R>({ key, value }: { key: string | null; value: R[CK] }): R[CK] | string {
+  // TODO: Do we need to activate editor when Delete, Backspace or other keys are pressed?
+  // These keys only make sense for a textbox editor. Activating a complex editor like a date-picker or a modal would not be intuitive
+  if (key === 'Delete' || key === 'Backspace') {
+    return '';
+  }
+  if (key === 'Enter') {
+    return value;
+  }
 
-  changeCommitted = false;
-  changeCanceled = false;
+  return key || value;
+}
 
-  private readonly editor = React.createRef<Editor>();
-  readonly state: Readonly<State> = { isInvalid: false };
+export default function EditorContainer<R, RK extends keyof R, CK extends keyof R = keyof R>({
+  rowIdx,
+  column,
+  rowData,
+  width,
+  height,
+  left,
+  top,
+  onCommit,
+  onCommitCancel,
+  scrollLeft,
+  scrollTop,
+  firstEditorKeyPress,
+  ...props
+}: EditorContainerProps<R, RK, CK>) {
+  const legacyEditorRef = useRef<Editor<{ [k: string]: R[CK] }>>(null);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const editorRef = useRef<any>(null);
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [value, setValue] = useState(() => getInitialValue({ key: firstEditorKeyPress, value: props.value }));
+  const prevScrollLeft = useRef(scrollLeft);
+  const prevScrollTop = useRef(scrollTop);
+  const { enableNewEditor, getRowMetaData, editor } = column;
+
+  const getInputNode = useCallback(() => {
+    return enableNewEditor
+      ? editorRef.current
+      : legacyEditorRef.current && legacyEditorRef.current.getInputNode();
+  }, [enableNewEditor]);
+
+  useEffect(() => {
+    const inputNode = getInputNode();
 
-  componentDidMount() {
-    const inputNode = this.getInputNode();
     if (inputNode instanceof HTMLElement) {
       inputNode.focus();
     }
     if (inputNode instanceof HTMLInputElement) {
```

Comment:
Just FYI, I believe there is a bug here for a long time. Please see my fix.

https://github.com/adazzle/react-data-grid/pull/1813/files#diff-bbe7676c3301843d12329fb8ff07e0f3R29

Reply:
@qili26 wdyt about the last commit? The issue is that we enable the editor on any key press. I think users should have some control over when to enable the editor and whether to select the value or not. I ended up removing `editorRef`, I think each editor should handle `keyDown` event and prevent default navigation if needed.

Reply:
I think removing the `editorRef` falls back to my implementation, which provides couple hooks for others to composite. Using context was just for passing the internal props directly to the hook so that when people are using it, they could just use like `useRDGEditor` (or like `useRDGKeys()`, etc..).